src/test/java/net/snowflake/client/jdbc/MultiStatementLatestIT.java

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (c) 2022 Snowflake Computing Inc. All right reserved.
+ */
+package net.snowflake.client.jdbc;
+
+import static org.junit.Assert.*;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import net.snowflake.client.category.TestCategoryStatement;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+/**
+ * MultiStatement integration tests for the latest JDBC driver. This doesn't work for the oldest
+ * supported driver. Revisit this tests whenever bumping up the oldest supported driver to examine
+ * if the tests still is not applicable. If it is applicable, move tests to MultiStatementIT so that
+ * both the latest and oldest supported driver run the tests.
+ */
+@Category(TestCategoryStatement.class)
+public class MultiStatementLatestIT extends BaseJDBCTest {
+  protected static String queryResultFormat = "json";
+
+  public static Connection getConnection() throws SQLException {
+    Connection conn = BaseJDBCTest.getConnection();
+    Statement stmt = conn.createStatement();
+    stmt.execute("alter session set jdbc_query_result_format = '" + queryResultFormat + "'");
+    stmt.close();
+    return conn;
+  }
+
+  @Test
+  public void testMultiStmtExecute() throws SQLException {
+    Connection connection = getConnection();
+    Statement statement = connection.createStatement();
+
+    statement.unwrap(SnowflakeStatement.class).setParameter("MULTI_STATEMENT_COUNT", 3);
+    String multiStmtQuery =
+        "create or replace temporary table test_multi (cola int);\n"
+            + "insert into test_multi VALUES (1), (2);\n"
+            + "select cola from test_multi order by cola asc";
+
+    boolean hasResultSet = statement.execute(multiStmtQuery);
+    // first statement
+    assertFalse(hasResultSet);
+    assertNull(statement.getResultSet());
+    assertEquals(0, statement.getUpdateCount());
+
+    // second statement
+    assertFalse(statement.getMoreResults());
```

Comment:
This should return `true` instead of `false` because there are more results available for fetching in the multistatement. Even though the current INSERT statement does not return a ResultSet object, there are more statements available which will return results.

This should return true because there is another SELECT statement afterwards which will return results. Basically should return true whenever there's statements to iterate through in the multistatement.

Reply:
[getMoreResults()](https://docs.oracle.com/javase/7/docs/api/java/sql/Statement.html#getMoreResults()) Returns true if the next result is a ResultSet object; _false if it is an update count_ or there are no more results

Reply:
@sfc-gh-igarish can you comment on this?

Reply:
No. Result means either resultset or update count. If any of that exist then getMoreResults() should return true.